internal/command/container.go

```diff
@@ -0,0 +1,393 @@
+package command
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/fyne-io/fyne-cross/internal/icon"
+	"github.com/fyne-io/fyne-cross/internal/log"
+	"github.com/fyne-io/fyne-cross/internal/volume"
+	"golang.org/x/sys/execabs"
+)
+
+const (
+	// fyneBin is the path of the fyne binary into the docker image
+	fyneBin = "/usr/local/bin/fyne"
+	// gowindresBin is the path of the gowindres binary into the docker image
+	gowindresBin = "/usr/local/bin/gowindres"
+)
+
+type ContainerRunner interface {
+	NewImageContainer(arch Architecture, OS string, image string) ContainerImage
+
+	Debug(v ...interface{})
+	GetDebug() bool
+}
+
+type AllContainerRunner struct {
+	ContainerRunner
+
+	Env  map[string]string // Env is the list of custom env variable to set. Specified as "KEY=VALUE"
+	Tags []string          // Tags defines the tags to use
+
+	vol volume.Volume
+
+	debug bool
+}
+
+type ContainerImage interface {
+	Cmd(vol volume.Volume, opts Options, cmdArgs []string) *execabs.Cmd
+	Run(vol volume.Volume, opts Options, cmdArgs []string) error
+	Prepare() error
+	Finalize(srcFile string, packageName string) error
+
+	GetArchitecture() Architecture
+	GetOS() string
+	GetID() string
+	GetTarget() string
+	GetEnv(string) (string, bool)
+	SetEnv(string, string)
+	SetMount(string, string)
+	AppendTag(string)
```

Comment:
Do we really need to add these methods as part of an interface ? Should be they removed in favor of a contract with the `AllContainerImage` struct ?

Reply:
That sounds doable. I was thinking I would have to change the behavior of some of them, but actually not at all.

Reply:
Remember that in Go it is idiomatic to remove `Get` prefix for simple getter calls.

Reply:
Are the Os/Arch part of the API at all? a `ContainerImage` is a generic construct that is created by the `ContainerRunner` calling some constructor passing in state, so this feels a little circular.

Reply:
After trying to remove them, it actually is more useful than expected as code use a `ContainerImage` to build figure out what command to execute. A `ContainerImage` is a representation of an Engine/OS/Architecture capability. I think they are for the moment useful where they are.